patroni/postgresql/connection.py

```diff
@@ -147,7 +147,7 @@ def get(self, name: str, kwargs_override: Optional[Dict[str, Any]] = None) -> Na
     def close(self) -> None:
         """Close all named connections from Patroni to PostgreSQL registered in the pool."""
         with self._lock:
-            if any(conn.close(True) for conn in self._connections.values()):
+            if all(conn.close(True) for conn in self._connections.values()):
```

Comment:
Well, indeed `any` stops after the first success, but the fix isn't correct.
We want to log if _any_ of the connections were closed.

Reply:
But we need to close all connections in this function for class ConnectionPool, right?
```
    def close(self) -> None:
        """**Close all named connections** from Patroni to PostgreSQL registered in the pool."""
        with self._lock:
            if any(conn.close(True) for conn in self._connections.values()):
                logger.info("**closed patroni connections to postgres**")
```

If we want to log if any of the connections were closed, I think we can set 'silent' to False to log it.

```
    def close(self, silent: bool = False) -> bool:
        """Close the psycopg connection to postgres.

        :param silent: whether the method should not write logs.

        :returns: ``True`` if ``psycopg`` connection was closed, ``False`` otherwise.``
        """
        ret = False
        if self._connection and self._connection.closed == 0:
            self._connection.close()
            if not silent:
                logger.info("closed patroni %s connection to postgres", self._name)
            ret = True
        self._connection = None
        return ret
```

Please check if below change is OK. Please tell me if there is any misunderstanding. 
```
    def close(self) -> None:
        """**Close all named connections** from Patroni to PostgreSQL registered in the pool."""
        with self._lock:
            if **all**(conn.close(**True**) for conn in self._connections.values()):
                logger.info("**closed patroni connections to postgres**")
```